owncloudData/src/test/java/com/owncloud/android/data/sharing/shares/repository/OCShareRepositoryTest.kt

```diff
@@ -20,496 +20,287 @@
-        OCShareRepository(localShareDataSource, remoteShareDataSource)
-
-    private val shares = arrayListOf(
-        OC_PUBLIC_SHARE,
-        OC_PRIVATE_SHARE,
-        OC_PRIVATE_SHARE,
-        OC_PUBLIC_SHARE
-    )
-
-    private val filePath = "/Images"
-    private val accountName = "admin@server"
-
-    @Test
-    fun refreshSharesFromNetworkOk() {
-        every { remoteShareDataSource.getShares(any(), any(), any(), any()) } returns shares
-
-        ocShareRepository.refreshSharesFromNetwork(filePath, accountName)
-
-        verify(exactly = 1) {
-            remoteShareDataSource.getShares(
-                remoteFilePath = filePath,
-                reshares = true,
-                subfiles = false,
-                accountName = accountName
-            )
-        }
+    private val localShareDataSource = mockk<LocalShareDataSource>(relaxUnitFun =  true)
+    private val remoteShareDataSource = mockk<RemoteShareDataSource>(relaxUnitFun = true)
+    private val ocShareRepository = OCShareRepository(localShareDataSource, remoteShareDataSource)
 
-        verify(exactly = 1) {
-            localShareDataSource.replaceShares(shares)
-        }
-    }
+    private val listOfShares = listOf(OC_PRIVATE_SHARE, OC_PUBLIC_SHARE)
+    private val filePath = OC_SHARE.path
+    private val password = "password"
+    private val permissions = OC_SHARE.permissions
+    private val expiration = RemoteShare.INIT_EXPIRATION_DATE_IN_MILLIS
 
     @Test
-    fun refreshSharesFromNetworkEmptyShares() {
-        every { remoteShareDataSource.getShares(any(), any(), any(), any()) } returns listOf()
-
-        ocShareRepository.refreshSharesFromNetwork(filePath, accountName)
-
-        verify(exactly = 1) {
-            remoteShareDataSource.getShares(
+    fun `insertPrivateShare inserts a private OCShare correctly`() {
+        every {
+            remoteShareDataSource.insert(
                 remoteFilePath = filePath,
-                reshares = true,
-                subfiles = false,
-                accountName = accountName
+                shareType = OC_PRIVATE_SHARE.shareType,
+                shareWith = OC_SHAREE.shareWith,
+                permissions = permissions,
+                name = "",
+                password = "",
+                expirationDate = expiration,
+                accountName = OC_ACCOUNT_NAME
             )
-        }
-
-        verify(exactly = 1) {
-            localShareDataSource.deleteSharesForFile(filePath, accountName)
-        }
-    }
+        } returns OC_PRIVATE_SHARE
 
-    @Test(expected = NoConnectionWithServerException::class)
-    fun refreshSharesFromNetworkNoConnection() {
-        every { remoteShareDataSource.getShares(any(), any(), any(), any()) } throws NoConnectionWithServerException()
+        every {
+            localShareDataSource.insert(OC_PRIVATE_SHARE)
+        } returns 1 // The result of this method is not used, so it can be anything
```

Comment:
It is better if you add the comment just above the block, like:
```suggestion
        // The result of this method is not used, so it can be anything
        every {
            localShareDataSource.insert(OC_PRIVATE_SHARE)
        } returns 1
```
So that we know the mocking we are about to read doesn't matter 😉. In this case it's ok to read it like this, but if it has several parameters we can get confused

Reply:
I've changed all comments anyway to keep the same structure